[Release] Run release script to change 3.0.0.dev0 to 2.9.0

architkulkarni opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?
Runs the following script from the release process to update the release numbers:

```
# Adapt the following to the correct version number for this release.
# This updates versions in Python and C++ files (This should, at a minimum, update python/ray/init.py, [build-docker.sh](http://build-docker.sh/), and src/ray/common/constants.h). (changes from python/ray/_private/utils.py, python/ray/tests/test_runtime_env.py, dashboard/client/package-lock.json should not be committed)
grep -l -R "3.0.0.dev0" * | grep -v "yaml" | grep -v "rst" | xargs -n 1 gsed -i 's/3.0.0.dev0/2.5.0/g'

# Note: python/ray/tests/test_runtime_env.py should NOT be changed!!
git checkout python/ray/tests/test_runtime_env.py

# This updates versions in java files
grep -l -R "2.0.0-SNAPSHOT" * | grep -v "yaml" | grep -v "rst" | xargs -n 1 gsed -i 's/2.0.0-SNAPSHOT/2.5.0/g'
```

I manually skipped the change to `python/ray/_private/utils.py` as indicated in the instruction.

<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

architkulkarni commented on python/requirements_compiled.txt:
```diff
@@ -1844,7 +1844,7 @@ querystring-parser==1.2.4
     #   tune-sklearn
     #   xgboost-ray
-raydp==1.7.0b20231020.dev0
+raydp==1.7.0b2022.9.0
```
I'm not sure how the command `grep -l -R "3.0.0.dev0" * | grep -v "yaml" | grep -v "rst" | xargs -n 1 gsed -i 's/3.0.0.dev0/2.9.0/g'` causes this change, but it does.  I will revert this change.  Any thoughts on how to fix the command?

aslonnie replied:
this should be reverted.

the `grep` thing is terribly broke imo...

architkulkarni commented on doc/source/ray-overview/pip_freeze_ray-py39-cpu.txt:
```diff
@@ -121,7 +121,7 @@ python-dotenv==1.0.0
 PyWavelets==1.4.1
 PyYAML==6.0.1
-ray @ file:///home/ray/ray-3.0.0.dev0-cp39-cp39-manylinux2014_x86_64.whl#sha256=85cf93d21b91ecef4cf6935194d03ca0c3f907d76267ecd5ad1c6fa6c768717b
+ray @ file:///home/ray/ray-2.9.0-cp39-cp39-manylinux2014_x86_64.whl#sha256=85cf93d21b91ecef4cf6935194d03ca0c3f907d76267ecd5ad1c6fa6c768717b
```
Do we need to revert this change, or wait until some later step in the process and then update the `sha` value?

aslonnie replied:
yes, please revert.

architkulkarni commented:
It looks like the release process instruction has not been updated.  I will use https://github.com/ray-project/ray/pull/41625 as a reference and update the release instruction.

aslonnie commented on doc/source/tune/examples/pbt_visualization/pbt_visualization.ipynb:
```diff
@@ -290,15 +290,15 @@
       ],
       "text/plain": [
-       "RayContext(dashboard_url='', python_version='3.8.13', ray_version='3.0.0.dev0', ray_commit='{{RAY_COMMIT_SHA}}', address_info={'node_ip_address': '127.0.0.1', 'raylet_ip_address': '127.0.0.1', 'redis_address': None, 'object_store_address': '/tmp/ray/session_2022-09-14_11-43-30_517456_2265/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2022-09-14_11-43-30_517456_2265/socket ...
+       "RayContext(dashboard_url='', python_version='3.8.13', ray_version='2.9.0', ray_commit='{{RAY_COMMIT_SHA}}', address_info={'node_ip_address': '127.0.0.1', 'raylet_ip_address': '127.0.0.1', 'redis_address': None, 'object_store_address': '/tmp/ray/session_2022-09-14_11-43-30_517456_2265/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2022-09-14_11-43-30_517456_2265/sockets/ray ...
```
revert these? I do not think they are really required.

but I am also not sure how this notebook is being used. shouldn't these notebook output be re-generated dynamically?

aslonnie commented on release/k8s_tests/run_gcs_ft_on_k8s.py:
```diff
@@ -25,7 +25,7 @@ class TestScenario(enum.Enum):
 if os.environ.get("RAY_IMAGE") is not None:
     ray_image = os.environ.get("RAY_IMAGE")
-elif ray.__version__ != "3.0.0.dev0":
+elif ray.__version__ != "2.9.0":
```
this should be reverted.

aslonnie commented on release/ray_release/tests/test_wheels.py:
```diff
@@ -49,20 +49,20 @@ def test_get_ray_wheels_url(remove_buildkite_env):
         branch="master",
         commit="1234",
-        ray_version="3.0.0.dev0",
+        ray_version="2.9.0",
```
all changes in this file should be reverted.

all these are unit test cases of `ray_release`.

aslonnie commented on release/ray_release/wheels.py:
```diff
@@ -158,7 +158,7 @@ def get_ray_wheels_url_from_local_wheel(ray_wheels: str) -> Optional[str]:
     The uploaded object will have local user and current timestamp encoded
     in the upload key path, e.g.:
-    "ubuntu_2022_01_01_23:59:99/ray-3.0.0.dev0-cp37-cp37m-manylinux_x86_64.whl"
+    "ubuntu_2022_01_01_23:59:99/ray-2.9.0-cp37-cp37m-manylinux_x86_64.whl"
```
also please revert these.

aslonnie commented on rllib/tests/backward_compat/checkpoints/v1.0/dqn_frozenlake_tf/policies/default_policy/rllib_checkpoint.json:
```diff
@@ -1 +1 @@
-{"type": "Policy", "checkpoint_version": "1.0", "ray_version": "3.0.0.dev0", "ray_commit": "5ae1638c8797518dc2177d7cf5d327cd9c32d706"}
\ No newline at end of file
+{"type": "Policy", "checkpoint_version": "1.0", "ray_version": "2.9.0", "ray_commit": "5ae1638c8797518dc2177d7cf5d327cd9c32d706"}
```
all the rllit tests and checkpoints stuff should be reverted.

architkulkarni commented:
@aslonnie thanks for the careful review.  I have updated the PR to match https://github.com/ray-project/ray/pull/40515, with the following exceptions:

New in 2.9.0
-> test_reporter.py.  You mentioned this should be fixed by using version in _version.py instead of hardcoding. This makes sense, I think we can just fix it on master and only fix it here if it breaks a test, but let me know if you think otherwise.

Changed from 2.8.0 to 2.9.0
-> the __init__.py update is now in _version.py instead.
